src/lib.rs

```diff
@@ -1163,4 +1177,37 @@ mod tests {
             Ok(())
         }
     }
+
+    #[cfg(feature = "arbitrary")]
+    mod arbitrary {
+        use crate::NonEmpty;
+        use arbitrary::{Arbitrary, Unstructured};
+
+        #[derive(Debug, Eq, PartialEq, arbitrary::Arbitrary)]
+        pub struct SimpleArbitrary(pub i32);
+
+        #[test]
+        fn test_arbitrary_empty_tail() -> arbitrary::Result<()> {
+            let mut u = Unstructured::new(&[1, 2, 3, 4]);
+            let ne = NonEmpty::<SimpleArbitrary>::arbitrary(&mut u)?;
+            assert!(!ne.is_empty());
+            assert_eq!(ne[0], SimpleArbitrary(67305985));
```

Comment:
I think it would be clearer for me if the assertions were:
```rust
assert!(ne.tail.is_empty());
assert_eq!(ne.head, SimpleArbitrary(67305985));
```

Reply:
I ended up doing a full `assert_eq!` check which is more simple IMO